src/users/dto/update-user-settings.dto.ts

```diff
@@ -18,13 +10,14 @@ export class UpdateUserSettingsDto {
   @IsNumber()
   readonly jobCount?: number;
 
-  @ApiProperty()
-  @IsArray()
-  readonly filters: FilterConfig[];
-
-  @ApiProperty()
-  @IsArray()
-  readonly conditions: ScientificCondition[];
+  @ApiProperty({
+    type: "object",
+    additionalProperties: { type: "array", items: {} },
+    required: false,
+  })
+  @IsOptional()
+  @IsObject()
+  readonly frontendSettings?: Record<string, unknown[]>;
```

Comment:
On second thought, can we call them just `extraSettings`?

Reply:
extraSettings are too vague. I strongly against naming it as extraSettings.

Reply:
ExtraSettings seems a bit too vague. I would kindly suggest reconsidering the name to something more specific.